### PR TITLE
feature/DEV-39-project-details

### PR DESCRIPTION
**Pull request by RC-3222**
Project details page and form (+ generic form number & date inputs)

Admin's POV
image

Generic user's POV
image

### CODE REVIEW

**Review by shev7 — APPROVED** [review]

File: src/modules/projects/api/update-project-mutation.ts [hunk 1]
```diff
@@ -2,6 +2,10 @@ import { gql, useMutation } from "@apollo/client";
 import { UpdateProjectInput } from "cv-graphql";
 import { UpdateProjectResult } from "./projects.types";
 import { GET_PROJECTS_QUERY } from "./get-projects-query";
+import { RouteParams } from "@/router";
```
Comment (AAKlimkov) [inline src/modules/projects/api/update-project-mutation.ts:6]: import order

File: src/modules/projects/components/project-details-content/project-details-content.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import { ProjectDetailsForm } from "../project-details-form";
+
+interface ProjectDetailsFormProps {
+    projectId: string;
+}
+
+const ProjectDetailsContent = ({ projectId }: ProjectDetailsFormProps) => {
+    const { project, loading } = useProjectQuery({ projectId });
+
+    if (!project && loading) return <Spin size="large" />;
```
Comment (AAKlimkov) [inline src/modules/projects/components/project-details-content/project-details-content.tsx]: should we really check !project for loading?

File: src/modules/projects/api/update-project-mutation.ts (hunk 1 above)
Comment (RC-3222) [reply src/modules/projects/api/update-project-mutation.ts:6]: fixed

File: src/modules/projects/components/project-details-content/project-details-content.tsx (hunk 2 above)
Comment (RC-3222) [reply src/modules/projects/components/project-details-content/project-details-content.tsx]: removed

File: src/modules/projects/api/update-project-mutation.ts [hunk 3]
```diff
@@ -19,7 +23,17 @@ export const UPDATE_PROJECT = gql`
...
+
     return useMutation<UpdateProjectResult, { project: UpdateProjectInput }>(UPDATE_PROJECT, {
-        refetchQueries: [GET_PROJECTS_QUERY],
+        refetchQueries: [
+            GET_PROJECTS_QUERY,
+            { query: GET_PROJECT_QUERY, variables: { projectId } },
+        ],
+        onError: (error) => {
+            showNotification("error", error.message);
+        },
```
Comment (savelievamareena) [inline src/modules/projects/api/update-project-mutation.ts:37]: maybe add onCompleted here to show notification that the project have been successfully updated?

**Review by savelievamareena — APPROVED** [review]

File: src/modules/projects/api/update-project-mutation.ts (hunk 3 above)
Comment (RC-3222) [reply src/modules/projects/api/update-project-mutation.ts:37]: well, haven't seen any "on success" notifications in example app (only "on error"), so why add those here?

**Pull request merged**